src/Jupyter/ConfigurationSource.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Quantum.IQSharp.Jupyter
+{
+
+    public interface IConfigurationSource
+    {
+        IDictionary<string, JToken> Configuration { get; }
+
+        void Persist();
+
+        private T GetOptionOrDefault<T>(string optionName, T defaultValue) =>
+            Configuration.TryGetValue(optionName, out var token)
+            ? token.ToObject<T>()
+            : defaultValue;
+
+        public BasisStateLabelingConvention BasisStateLabelingConvention =>
+            GetOptionOrDefault("dump.basisStateLabelingConvention", BasisStateLabelingConvention.LittleEndian);
+
+        public bool TruncateSmallAmplitudes =>
+            GetOptionOrDefault("dump.truncateSmallAmplitudes", false);
+
+        public double TruncationThreshold =>
```

Comment:
so they allow implementation in interfaces now?

Reply:
That's a new C# 8 feature, default properties and methods... as far as I understand, intended to provide traits-style programming. I'm still wrapping my head around the concept of a _private_ method in an interface, that takes some getting used to.